src/data-type.js

```diff
@@ -1305,10 +1305,30 @@ const TYPE = module.exports.TYPE = {
       if (value instanceof Date) {
         return value;
       }
-      value = Date.parse(value);
-      if (isNaN(value)) {
+      var dateValue = Date.parse(value);
+      if (!isNaN(dateValue)) {
+        return dateValue;
+      }
+      var timespan = (function(input) {
+        var regex = /^([0-9]{1}|(?:0[0-9]|1[0-9]|2[0-3])+):([0-5]?[0-9])(?::([0-5]?[0-9])(?:\.(\d{1,9}))?)?$/;
+        var result = (regex.exec(input) || []).slice(1, 5);
```

Comment:
Skipping the `slice` here would eliminate an array allocation.

Changing `regex.exec(input) || []` to `regex.exec(input) || [0,0,0,0,0]` would simplify the code below because `|| 0` wouldn't be needed.